### ISSUE TITLE
REST API bundle-products/:sku/options/all always return is not authorized

### ISSUE THREAD

**Issue by feelinc** [magento/magento2#9468]
### Preconditions
1. PHP 7.0
2. Magento 2.1.6

### Steps to reproduce
1. Web API Security -> Allow Anonymous Guest Access: Yes
2. Try GET "/rest/V1/bundle-products/:sku/options/all" using Postman

### Expected result
1. Never see a correct result

### Actual result
`{
  "message": "Consumer is not authorized to access %resources",
  "parameters": {
    "resources": "Magento_Catalog::products"
  }
}`

![selection_299](https://cloud.githubusercontent.com/assets/345562/25602156/c3200e94-2f1b-11e7-8603-ddfe4f90d611.jpg)

**Comment by ericahoekerd** [comment]
This is related to a very very old bug, it is back now in 2.16

https://github.com/magento/magento2/issues/5330

**Comment by misha-kotov** [comment]
No, this is a different issue. Allowing anonymous Web API access in the admin should unlock the REST catalog calls and not require the auth token. @feelinc can you try this endpoint? 
`V1/products/:sku`
This works for me after I turn on anonymous access and flush the cache.

**Comment by feelinc** [comment]
The **V1/products/:sku**, it's working correctly with anonymous access.

I'm currently testing to develop a android app using the Magento 2 REST API, following is what I had so far with it -> https://www.youtube.com/watch?v=6pLKfSFg9WI

The **V1/bundle-products/:sku/options/all** still having the same issue after flushing the cache, below are the cli commands I use : 
```
rm -rf var/di var/generation
php bin/magento cache:clean
php bin/magento cache:flush
php bin/magento setup:upgrade
php bin/magento setup:di:compile
php bin/magento indexer:reindex
```

I think it's not an urgent issue, because related with anonymous access only. Similar with **V1/products/attributes** endpoint.

**Comment by johnpdavis** [comment]
On Version 2.1.2:
GET    /V1/bundle-products/:productSku/children      returns the same authorization error. 
GET    /V1/configurable-products/:sku/children       works great!

Hoping this is fixed soon.

**Comment by magento-engcom-team** [comment]
@feelinc, thank you for your report.
We've created internal ticket(s) MAGETWO-80820 to track progress on the issue.

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to 2.2-develop branch. Will be available with upcoming patch release
